Open/close modals using nav and selected by user icon appears in appropriate box

apyd opened the pull request:
- Selecting nav item opens appropriate modal
- Closing modal on 'x'
- Closing modal when click outside of modal
- Closing previously opened modal if new modal was opened by using nav items

apyd committed:
Open/close modals when using nav links

- Selecting nav item opens appropriate modal
- Closing modal on 'x'
- Closing modal when click outside of modal
- Closing previously opened modal if new modal was opened by using nav items

apyd committed:
User selection appears in appropriate box

- User selection appears in appropriate box
- By default question mark icon appears within box (before any user selection)

Mikailus commented on css/style.css:
```diff
@@ -151,4 +163,47 @@
+    background-color: white;
+    z-index: 20;
+    margin: 0 auto;
+    z-index: 2;
```
Duplicated property

Mikailus commented on css/style.css:
```diff
@@ -151,4 +163,47 @@
+    margin: 0 auto;
+    z-index: 2;
+}
+.modal--close {
```
I think you should rethink naming `modal--close` as it refers in your code to a button, not modal itself. If you meant that it is a block being `modal close`, it should have only one hyphen (but actually it should be treated as an element of a modal, not independent block). Two hyphens means that it is a modifier for a modal

Mikailus commented on index.html:
```diff
@@ -16,13 +16,42 @@
         <li class="nav__item">
-          <a href="#" class="nav__link">Previous moves</a>
+          <a href="#" class="nav__link">Previous moves</a></li>
         </li>
```
Should this `li` persist?

Mikailus commented on index.html:
```diff
@@ -16,13 +16,42 @@
         <li class="nav__item">
-          <a href="#" class="nav__link">Rules</a></li>
+          <a href="#" class="nav__link">Rules</a>
+          
```
Blank space

Mikailus commented on css/style.css:
```diff
@@ -15,19 +15,29 @@
     font-size: 1.5em;
     display: block;
     padding: 15px;
+    z-index: 5;
```
Does it need `z-index` property as it is set to `nav__list`?

Mikailus commented on css/style.css:
```diff
@@ -15,19 +15,29 @@
 .nav__item {
     border-bottom: 1.5px solid  #F1C40F;
     margin: 0 5px;
+    z-index: 5;
```
Does it need `z-index` property as it is set to `nav__list`?

Mikailus commented on css/style.css:
```diff
@@ -151,4 +163,47 @@
+    height: 830px;
+    top: 8%;
+    left: 28%;
+    background-color: white;
```
Previously you used hex code for colors, it is a good pattern to use only one type of coding colors - if you use hex, then use it across application (with exception to rgba)

Mikailus commented on index.html:
```diff
@@ -16,13 +16,42 @@
+      <div class="modal-bg">
+        <div class="modal">
+          <div class="modal__content">
+            <h2 class="modal__header">Rules</h2>
```
I think the class name should be `modal__heading` here (https://wikidiff.com/heading/header)

Mikailus commented on index.html:
```diff
@@ -16,13 +16,42 @@
+                <h2 class="modal__header">Author</h2>
+                <p>This is test modal for author</p>
+              </div>
+            </div>
```
This line and below - indentation

Mikailus commented on js/script.js:
```diff
@@ -0,0 +1,48 @@
+        closeModal();
+}
+    openedModal = e.srcElement.elementNumber;    
+    modals[openedModal].style.visibility = 'visible';
```
Does `e.srcElement === modals[openedModal]`?

Mikailus replied:
The first one is a link

Mikailus commented on js/script.js:
```diff
@@ -0,0 +1,48 @@
+
+function closeModalWhenClickOutside(e) {
+    if(e.target == modals[openedModal]) {
+    for(i = 0; i < modals.length; i++) {
```
Indentation

Mikailus commented on js/script.js:
```diff
@@ -0,0 +1,48 @@
+};
+
+function closeModalWhenClickOutside(e) {
+    if(e.target == modals[openedModal]) {
```
Doesn't this statement mean that when the opened modal is the target of a click event, it should be closed?

Mikailus commented on js/script.js:
```diff
@@ -0,0 +1,48 @@
+var links = document.getElementsByClassName('nav__link');
+var closeButtons = document.getElementsByClassName("modal--close")
+var choiceOptions = document.getElementsByClassName('choice-container__icon');
+var userChoice;
```
Does this variable need to be declared here?

Mikailus commented on js/script.js:
```diff
@@ -0,0 +1,48 @@
+}
+
+function selectedOption(e) {
+    userChoice = choiceOptions[e.currentTarget.optionNumber];
```
Does `e.currentTarget === choiceOptions[e.currentTarget.optionNumber]`?

apyd committed:
Showing nav item as active and applied code review advices

1. Applied all changes mentioned in code review + refactoring of current code:
- class name change - from 'modal header' to 'modal heading'
- changed class for modal close button
- removed variables responsible for storing opened item - using now events instead of it
- removed unecessary z-indexes, indentations

2. Nav item is active when modal is opened

Mikailus commented on css/style.css:
```diff
@@ -15,19 +15,29 @@
+    z-index: 5;
 }
 
+.nav__item:active {
```
``` .nav__item:active, .nav__link:active```
